stablehlo/reference/Ops.cpp

```diff
@@ -156,6 +156,27 @@ Tensor makeSplat(ShapedType type, const Element &initValue) {
   return result;
 }
 
+SmallVector<Tensor> split(const Tensor &input, int64_t groupSize,
+                          Axis splitDimension, MLIRContext *context) {
+  Sizes splitInputShape(input.getShape());
+  splitInputShape[splitDimension] /= groupSize;
```

Comment:
Let's also check and error out if the split dimension size is not divisible by `numResults`.